pkg/client/elasticsearch/query.go

```diff
@@ -215,6 +219,21 @@ func getFloat(v interface{}) (float64, error) {
 	}
 }
 
+func parseResponseBodyError(decodedResponseBody map[string]interface{}) (map[string]interface{}, error) {
```

Comment:
Would it not be simpler to to have a struct and parse into that?

Something like 
```
// ErrorResponse is an Elasticsearch error response.
type ErrorResponse struct {
	Status int `json:"status"`
	Error  struct {
		CausedBy struct {
			Reason string `json:"reason"`
			Type   string `json:"type"`
		} `json:"caused_by"`
		Reason     string `json:"reason"`
		Type       string `json:"type"`
		StackTrace string `json:"stack_trace,omitempty"`
		RootCause  []struct {
			Reason string `json:"reason"`
			Type   string `json:"type"`
		} `json:"root_cause"`
	} `json:"error"`
}
```
Similar to what we have in ECK https://github.com/pebrc/cloud-on-k8s/blob/cda3509fa42e63d64003de3fbdc48464360a9b7c/pkg/controller/elasticsearch/client/error.go#L46

Reply:
We can simplify the struct to only contain the fields we are interested in.

Reply:
makes sense, I can make that change